superset-frontend/src/SqlLab/components/SaveQuery/index.tsx

```diff
@@ -178,7 +193,9 @@ export default function SaveQuery({
         onHide={() => setShowSaveDatasetModal(false)}
         buttonTextOnSave={t('Save & Explore')}
         buttonTextOnOverwrite={t('Overwrite & Explore')}
-        datasource={getDatasourceAsSaveableDataset(query)}
+        datasource={
+          getDatasourceAsSaveableDataset(query) as any as ISaveableDatasource
```

Comment:
why are casting as `any` and then `ISaveableDatasource` here

Reply:
oh it needed this custom type cast but no longer needed. I reverted this change.